cli/internal/typer/orchestrator.go

```diff
@@ -0,0 +1,150 @@
+package typer
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/rudderlabs/rudder-iac/cli/internal/app"
+	"github.com/rudderlabs/rudder-iac/cli/internal/logger"
+	"github.com/rudderlabs/rudder-iac/cli/internal/typer/generator/core"
+	"github.com/rudderlabs/rudder-iac/cli/internal/typer/generator/platforms/kotlin"
+	"github.com/rudderlabs/rudder-iac/cli/internal/typer/plan"
+)
+
+var (
+	orchestratorLog = logger.New("typer.orchestrator")
+)
+
+// Orchestrator coordinates the RudderTyper code generation process
+type Orchestrator struct {
```

Comment:
Let's actually rename this to `RudderTyper`, as mentioned in the design doc, to make it clear that this is the entry point for the typer logic